jenkins/jobs/dsl/java_reference_application_jobs.groovy

```diff
@@ -540,3 +541,87 @@ deployJobToProdB.with {
         )
     }
 }
+
+owaspDepCheckJob.with {
+    description("This job uses the OWASP Dependency Check plugin to check for CVEs in the application dependencies.")
+    wrappers {
+        preBuildCleanup()
+        injectPasswords()
+        maskPasswords()
+        sshAgent("adop-jenkins-master")
+    }
+    scm {
+        git {
+            remote {
+                url(referenceAppGitUrl)
+                credentials("adop-jenkins-master")
+            }
+            branch("*/master")
+        }
+    }
+    environmentVariables {
+        env('WORKSPACE_NAME', workspaceFolderName)
+        env('PROJECT_NAME', projectFolderName)
+    }
+    label("java8")
+
+    steps {
+        maven {
+            goals('clean install -DskipTests')
+            mavenInstallation("ADOP Maven")
+        }
+    }
+
+    configure { project ->
+        project / 'builders' / 'org.jenkinsci.plugins.DependencyCheck.DependencyCheckBuilder'(plugin: 'dependency-check-jenkins-plugin@1.4.4') {
+            skipOnScmChange false
+            skipOnUpstreamChange false
+            scanpath ''
+            outdir ''
+            datadir ''
+            suppressionFile ''
+            zipExtensions ''
+            isAutoupdateDisabled false
+            isVerboseLoggingEnabled false
+            includeHtmlReports false
+            useMavenArtifactsScanPath false
+        }
+    }
+
+    configure { project ->
```

Comment:
We can replace this with DSL, instead of just XML fallback https://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.helpers.publisher.PublisherContext.dependencyCheck